File: backend/app/routers/medical_records.py
from fastapi import APIRouter, Depends

from app.crud import medical_record as crud
from app.dependencies import DbSession, get_current_admin, get_current_doctor
from app.exceptions import ForbiddenError, NotFoundError
from app.schemas.medical_record import (
    MedicalRecordCreate,
    MedicalRecordResponse,
    MedicalRecordUpdate,
)

router = APIRouter(tags=["病历"])


@router.post("/api/v1/visits/{visit_id}/medical-record", response_model=MedicalRecordResponse, summary="创建病历 [需JWT]")
def create_medical_record(
    visit_id: int,
    schema: MedicalRecordCreate,
    db: DbSession,
    doctor=Depends(get_current_doctor),
):
    return crud.create_medical_record(db, visit_id, schema, doctor.id)


@router.get("/api/v1/visits/{visit_id}/medical-record", response_model=MedicalRecordResponse, summary="查看病历")
def get_medical_record(visit_id: int, db: DbSession):
    record = crud.get_medical_record(db, visit_id)
    if not record:
        raise NotFoundError("病历不存在")
    return record


@router.put("/api/v1/medical-records/{record_id}", response_model=MedicalRecordResponse, summary="更新病历 [仅admin]")
def update_medical_record(
    record_id: int,
    schema: MedicalRecordUpdate,
    db: DbSession,
    _=Depends(get_current_admin),
):
    return crud.update_medical_record(db, record_id, schema)

File: doctor-admin/src/api/medicalRecords.ts
import api from './index'

export interface MedicalRecord {
  id: number
  visit_id: number
  treatment_types: string[]
  prescription: string | null
  acupuncture_points: string[]
  tongue_diagnosis: string | null
  pulse_diagnosis: string | null
  physical_signs: string | null
  diagnosis: string | null
  syndrome_analysis: string | null
  treatment_plan: string | null
  notes: string | null
  created_at: string
}

export interface MedicalRecordCreate {
  treatment_types: string[]
  prescription?: string
  acupuncture_points?: string[]
  tongue_diagnosis?: string
  pulse_diagnosis?: string
  physical_signs?: string
  diagnosis?: string
  syndrome_analysis?: string
  treatment_plan?: string
  notes?: string
}

export const getMedicalRecord = (visitId: number) =>
  api.get<MedicalRecord>(`/api/v1/visits/${visitId}/medical-record`)

export const createMedicalRecord = (visitId: number, data: MedicalRecordCreate) =>
  api.post<MedicalRecord>(`/api/v1/visits/${visitId}/medical-record`, data)

export const updateMedicalRecord = (id: number, data: Partial<MedicalRecordCreate>) =>
  api.put<MedicalRecord>(`/api/v1/medical-records/${id}`, data)

export const addLabResult = (visitId: number, data: object) =>
  api.post(`/api/v1/visits/${visitId}/lab-results`, data)
